### PR TITLE
chore(ct): remove `suite` dependency by connecting dependency graphs at read time, not write time

### PR DESCRIPTION
**Pull request by Skn0tt**
Broken out of https://github.com/microsoft/playwright/pull/31727 as per @dgozman's [request](https://github.com/microsoft/playwright/pull/31727#discussion_r1685793229).

The PR goal is to remove the `suite` argument from the Component testing's Vite Plugin. `suite` is used to enrich Vite's dependency graph with information about dependencies between test suites and helper files. It essentially merges the Vite graph with the `compilationCache.ts > fileDependencies` graph, and then writes the result back into `compilationCache.ts > externalDependencies`.

By refactoring this to make the connection on the reading end in `collectAffectedTestFiles`, we can drop the `suite` parameter.

We didn't yet have a test that depended on the dependency graph being connected correctly between `fileDependencies` and `externalDepedencies`, so I've [extended an existing test](https://github.com/microsoft/playwright/commit/53a539938b8b90d4ffdce7783e340232dfb02a18) to capture that.

### CODE REVIEW

File: packages/playwright-ct-core/src/vitePlugin.ts [hunk 1]
```diff
@@ -48,8 +48,8 @@ export function createPlugin(): TestRunnerPlugin {
       configDir = configDirectory;
     },
 
-    begin: async (suite: Suite) => {
-      const result = await buildBundle(config, configDir, suite);
+    begin: async () => {
```
Comment (dgozman) [inline packages/playwright-ct-core/src/vitePlugin.ts:51]: Let's keep the unused `Suite` parameter here to mirror the `Plugin` interface.

File: packages/playwright/src/transform/compilationCache.ts [hunk 2]
```diff
@@ -218,9 +218,15 @@ export function collectAffectedTestFiles(dependency: string, testFileCollector:
     if (deps.has(dependency))
       testFileCollector.add(testFile);
   }
-  for (const [testFile, deps] of externalDependencies) {
-    if (deps.has(dependency))
-      testFileCollector.add(testFile);
+  for (const [importingFile, deps] of externalDependencies) {
+    if (deps.has(dependency)) {
+      testFileCollector.add(importingFile);
```
Comment (dgozman) [inline packages/playwright/src/transform/compilationCache.ts:223]: We should only add `importingFile` if it is a test file. That's what the line 215 checks for.

File: packages/playwright/src/transform/compilationCache.ts [hunk 3]
```diff
@@ -218,9 +218,15 @@ export function collectAffectedTestFiles(dependency: string, testFileCollector:
...
+
+      for (const [testFile, deps] of fileDependencies) {
```
Comment (dgozman) [inline packages/playwright/src/transform/compilationCache.ts:225]: nit: can we avoid shadowing the `deps` variable? I find it harder to read.

File: packages/playwright/src/transform/compilationCache.ts [hunk 4]
```diff
@@ -218,9 +218,15 @@ export function collectAffectedTestFiles(dependency: string, testFileCollector:
     if (deps.has(dependency))
       testFileCollector.add(testFile);
   }
-  for (const [testFile, deps] of externalDependencies) {
-    if (deps.has(dependency))
-      testFileCollector.add(testFile);
+  for (const [importingFile, deps] of externalDependencies) {
```
Comment (dgozman) [inline packages/playwright/src/transform/compilationCache.ts:221]: Let's also update `dependenciesForTestFile` function.

File: packages/playwright/src/transform/compilationCache.ts (hunk 3 above)
Comment (Skn0tt) [reply packages/playwright/src/transform/compilationCache.ts]: done!

File: packages/playwright-ct-core/src/vitePlugin.ts (hunk 1 above)
Comment (Skn0tt) [reply packages/playwright-ct-core/src/vitePlugin.ts]: done

File: packages/playwright/src/transform/compilationCache.ts (hunk 4 above)
Comment (Skn0tt) [reply packages/playwright/src/transform/compilationCache.ts]: I've updated this in https://github.com/microsoft/playwright/pull/31794/commits/72c446d8ad202dc60517c99be6755231b2352885 and verified manually that it works correctly, but i'm unsure how to add a test for it. The problem is that the function is only ever used to update the file watcher, and it's hard to observe that. Where would you add a test for this?

**Review by dgozman — APPROVED** [review]

File: packages/playwright/src/transform/compilationCache.ts (hunk 2 above)
Comment (dgozman) [reply packages/playwright/src/transform/compilationCache.ts]: This is not a performance-critical place, so premature optimization is not needed. Overall, we are serious about performance on the critical path, and in general in the core.

File: packages/playwright/src/transform/compilationCache.ts (hunk 4 above)
Comment (dgozman) [reply packages/playwright/src/transform/compilationCache.ts]: I think we can live without a test for this for now.

**Pull request merged**